packages/app-builder/src/components/ui/user/ejectable/table.tsx

```diff
@@ -49,62 +71,122 @@ const DeleteButton = ({ disabled, typeName, id }: { disabled?: boolean; typeName
   )
 }
 
+const shouldDisplay = ([key, value]: [string, any | undefined | null], subobjectNames: string[]) => {
+  const isHidden = key.startsWith('_')
+  const isId = key === 'id'
+  const isArray = value && Array.isArray(value)
+  const isSubObject = isArray
+    ? value.length > 0 && subobjectNames.includes(value[0].__typename)
+    : value && notEmpty(value) && subobjectNames.includes(value.__typename)
+
+  return !isHidden && !isId && !isSubObject
+}
+
 export const Table = forwardRef<HTMLDivElement, TableProps & { disabled?: boolean }>(
   ({ typeName, editPageId, showControls, disabled, ...props }, ref) => {
     const { data, loading } = useObjectList(typeName)
+    const { available: deletionAvailable } = useObjectDelete(typeName)
+    const { available: updateAvailable } = useObjectUpdate(typeName)
+    const subobjects = useSubObjects(typeName)
+    const { specials } = useObjectSpecials(typeName)
     const history = useHistory()
+    const { context, appId } = usePageId()
+
+    const subobjectNames = subobjects.data.map(({ object: { name } }) => name)
+
     const rows =
       data &&
       typeName &&
       data.map((row) => {
-        const cells = Object.entries(row)
+        let cells = Object.entries(row)
+          .filter((entry) => shouldDisplay(entry, subobjectNames))
           .map(([label, value]) => ({
             label: uppercaseSentence(label),
-            value: typeof value === 'object' ? undefined : value,
-            children: typeof value === 'object' ? <ObjectTableCell {...value} /> : undefined,
+            value: (typeof value === 'object' ? undefined : value) as string,
+            children: typeof value === 'object' ? <ObjectTableCell obj={value} /> : undefined,
             narrowTable: {
               showLabel: true,
             },
           }))
-          .filter((cell) => {
-            return !cell.label.startsWith('__') && cell.label.toLowerCase() !== 'id'
-          })
 
-        if (!showControls) {
-          return { cells }
-        }
+        const specialsAndSubobjects = [
+          ...specials.map(({ name }) => ({ name, label: uppercaseSentence(name) })),
+          ...subobjects.data.map(({ object }) => ({
+            name: object.name,
+            label: `View ${uppercaseSentence(object.name)}s`,
+          })),
+        ]
 
-        return {
-          cells: [
-            ...cells,
-            {
-              label: 'Edit',
+        cells = [
```

Comment:
Just a general comment but the table logic is a bit hard to follow. Feels like it could get broken down a bit into some functions or similar. Just a thought though...